pkg/builder/webhook.go

```diff
@@ -65,6 +66,12 @@ func (blder *WebhookBuilder) For(apiType runtime.Object) *WebhookBuilder {
 	return blder
 }
 
+// WithMutationHandler takes an admission.Handler, a MutatingWebhook will be wired for it.
+func (blder *WebhookBuilder) WithMutationHandler(h admission.Handler) *WebhookBuilder {
```

Comment:
Its super not obvious when and why to use this versus the defaulter and validator and almost guaranteed to lead to questions

Reply:
Can we leave it as "more advanced use cases where you want control over the construction of patches"?

Reply:
Or we could explicitly add the use case of go types that might be in a different version than the CRD.

Reply:
Please write something along the lines of that this is a low-level plug point for advanced users that need more control over how exactly the patch is being constructed and if in doubt, people should always use `WithDefaulter` instead.

Reply:
Added